### PR TITLE
Update `polaris.shopify.com` and `stylelint-polaris` config categories for new token groups

### LINKED ISSUE #8946: Update `polaris.shopify.com` and `stylelint-polaris` config categories for new token groups
Update `polaris.shopify.com` and `stylelint-polaris` config categories for `shape`, `colors`, and `depth` to `border`, `color`, and `shadow`.

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Now that our new token groups (`border`, `color`, `shadow`) are stable we want to encourage folks to start using them and update all of our documentation to reflect those changes.

### WHAT is this pull request doing?

Updates `polaris.shopify.com` and `stylelint-polaris` config categories to new token groups.

### Tophatting 🎩

Ensuring any legacy token groups are changed to new ones:  

- `shape` ➡️ `border` 
- `colors` ➡️ `color`
- `depth` ➡️ `shadow`

#### Where to look 👀
- http://localhost:3000/tokens/color
- http://localhost:3000/tokens/border
- http://localhost:3000/tools/stylelint-polaris#rules
- http://localhost:3000/tools/stylelint-polaris/rules
- All changed rule sub pages (border, color, shadow) ex. http://localhost:3000/tools/stylelint-polaris/rules/border-custom-property-disallowed-list

### CODE REVIEW

File: polaris.shopify.com/content/tools/stylelint-polaris/index.md [hunk 1]
```diff
@@ -82,32 +82,31 @@ npx @shopify/polaris-migrator styles-insert-stylelint-disable <path>
...
 - [Conventions](/tools/stylelint-polaris/rules#conventions)
-- [Depth](/tools/stylelint-polaris/rules#depth)
 - [Layout](/tools/stylelint-polaris/rules#layout)
-- [Legacy](/tools/stylelint-polaris/rules#legacy)
 - [Media queries](/tools/stylelint-polaris/rules#media-queries)
 - [Motion](/tools/stylelint-polaris/rules#motion)
-- [Shape](/tools/stylelint-polaris/rules#shape)
-- [Spacing](/tools/stylelint-polaris/rules#spacing)
+- [Shadow](/tools/stylelint-polaris/rules#shadow)
+- [Space](/tools/stylelint-polaris/rules#space)
```
Comment (lgriffee) [inline polaris.shopify.com/content/tools/stylelint-polaris/index.md:92]:
```suggestion
- [Spacing](/tools/stylelint-polaris/rules#spacing)
```
Revert this change will move to Alex's PR

File: polaris.shopify.com/src/components/TokensPage/TokensPage.tsx [hunk 2]
```diff
@@ -9,14 +9,13 @@ import Page from '../Page';
...
     | 'breakpoints'
-    | 'colors'
-    | 'depth'
+    | 'color'
     | 'font'
     | 'motion'
     | 'shadow'
-    | 'shape'
-    | 'spacing'
+    | 'space'
```
Comment (lgriffee) [inline polaris.shopify.com/src/components/TokensPage/TokensPage.tsx:18]:
```suggestion
    | 'spacing'
```
Revert handled in Alex's PR

File: polaris.shopify.com/src/components/TokensPage/TokensPage.tsx [hunk 3]
```diff
@@ -29,24 +28,20 @@ export type NavItem = {
...
   },
   {
-    title: 'Spacing',
-    url: `/tokens/spacing`,
-  },
-  {
-    title: 'Depth',
-    url: `/tokens/depth`,
+    title: 'Space',
+    url: `/tokens/space`,
```
Comment (lgriffee) [inline polaris.shopify.com/src/components/TokensPage/TokensPage.tsx]:
```suggestion
    title: 'Spacing',
    url: `/tokens/spacing`,
```

Revert handled in Alex's PR

File: polaris.shopify.com/src/components/TokensPage/TokensPage.tsx [hunk 4]
```diff
@@ -84,14 +79,13 @@ function TokensPage({tokenGroup}: Props) {
...
     breakpoints: tokensToFilteredArray(filter, allTokens.breakpoints),
-    colors: tokensToFilteredArray(filter, allTokens.colors),
-    depth: tokensToFilteredArray(filter, allTokens.depth),
+    color: tokensToFilteredArray(filter, allTokens.color),
     font: tokensToFilteredArray(filter, allTokens.font),
     motion: tokensToFilteredArray(filter, allTokens.motion),
     shadow: tokensToFilteredArray(filter, allTokens.shadow),
-    shape: tokensToFilteredArray(filter, allTokens.shape),
-    spacing: tokensToFilteredArray(filter, allTokens.spacing),
+    space: tokensToFilteredArray(filter, allTokens.space),
```
Comment (lgriffee) [inline polaris.shopify.com/src/components/TokensPage/TokensPage.tsx:88]:
```suggestion
    spacing: tokensToFilteredArray(filter, allTokens.spacing),
```
Revert handled in Alex's PR

**Comment by lgriffee** [comment]
@martenbjork Do we need to put any re-directs in place if we are updating URLs in this PR (specifically the token page default ex: `tokens/colors` ➡️ `tokens/color`?

File: .changeset/khaki-penguins-obey.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'polaris.shopify.com': minor
+---
+
+Update `polaris.shopify.com` and `stylelint-polaris` config categories for new token groups
```
Comment (lgriffee) [inline .changeset/khaki-penguins-obey.md]:
```suggestion
---
'@shopify/stylelint-polaris': major
'polaris.shopify.com': minor
---

Update `polaris.shopify.com` and `stylelint-polaris` config categories for new token groups
```

File: .changeset/khaki-penguins-obey.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/stylelint-polaris': major
```
Comment (lgriffee) [inline .changeset/khaki-penguins-obey.md]: Just want to double check this should be major?

Comment (lgriffee) [reply .changeset/khaki-penguins-obey.md]: I know it looks like I did a lot in that file (`stylelint-polaris/index.js`) but really I only renamed the old token group categories to the new ones (and also put them in alpha order).

File: polaris.shopify.com/pages/tools/stylelint-polaris/rules/[rule].tsx [hunk 7]
```diff
@@ -11,35 +11,33 @@ import {parseMarkdown} from '../../../../src/utils/markdown.mjs';
...
   'media queries':
     'Polaris [breakpoint sass variables](/tokens/breakpoints#sass-variables)',
+  motion: 'Polaris [motion tokens](/tokens/motion)',
+  shadow: 'Polaris [depth tokens](/tokens/shadow)',
+  spacing: 'Polaris [spacing tokens](/tokens/spacing)',
   'z-index': 'Polaris [z-index tokens](/tokens/z-index)',
-  layout: 'Polaris [layout components](/components)',
-  legacy: 'Polaris [components](/components) or [tokens](/tokens)',
+  typography:
+    'the [text component](/components/typography/text) or [font tokens](/tokens/font)',
```
Comment (aaronccasanova) [inline polaris.shopify.com/pages/tools/stylelint-polaris/rules/[rule].tsx]:
```suggestion
    'Polaris [text component](/components/typography/text) or [font tokens](/tokens/font)',
```

Not sure if this is accurate. Feel free to resolve.

**Review by aaronccasanova — APPROVED** [review]

File: .changeset/khaki-penguins-obey.md (hunk 6 above)
Comment (aaronccasanova) [reply .changeset/khaki-penguins-obey.md]: These updates shouldn't cause new CI failures. I think a `minor` upgrade would be fine here.

**Comment by martenbjork** [comment]
> @martenbjork Do we need to put any re-directs in place if we are updating URLs in this PR (specifically the token page default ex: `tokens/colors` ➡️ `tokens/color`?

I saw you put one in next.config.js — super!

For any other URLs that have changed: It's not required per se, but it's always a good idea to add redirects for any URL that might have changed. It helps Google find the new URLs and any outdated links will keep on working.

**Review by martenbjork — APPROVED** [review]

File: .changeset/khaki-penguins-obey.md (hunk 6 above)
Comment (lgriffee) [inline .changeset/khaki-penguins-obey.md]:
```suggestion
'@shopify/stylelint-polaris': minor
```

**Comment by lgriffee** [comment]
> For any other URLs that have changed: It's not required per se, but it's always a good idea to add redirects for any URL that might have changed. It helps Google find the new URLs and any outdated links will keep on working.

@martenbjork Perfect! I've added redirects for all other URLs that have changed and tophatted that they all work as expected.
